src/modeler/AutoRest.Swagger/Model/ServiceDefinitionMetadata.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Utilities;
+using System;
+using System.Collections.Generic;
+
+namespace AutoRest.Swagger.Model
+{
+    /// <summary>
+    /// Describes whether a service definition has already been processed and merged.
+    /// This can influence ARM validation (e.g. one would not expect an operations API on individual OpenAPI definitions).
+    /// </summary>
+    public enum ServiceDefinitionMergeState
```

Comment:
Please add as much detailed information as you can here explaining what "Before" & "After" means and in what context like be explicit that this only works with markdown structured mechanism